src/notation/qml/MuseScore/NotationScene/internal/DynamicPopup.qml

```diff
@@ -0,0 +1,213 @@
+import QtQuick 2.15
+import QtQuick.Layouts 1.15
+
+import Muse.Ui 1.0
+import Muse.UiComponents 1.0
+import MuseScore.NotationScene 1.0
+
+StyledPopupView {
+    id: root
+
+    property NavigationSection notationViewNavigationSection: null
+    property int navigationOrderStart: 0
+    property int navigationOrderEnd: dynamicsNavPanel.order
+
+    property int buttonHeight: 30
+
+    property int currentPage: 0
+    property bool isHairpinPage: currentPage == 6
```

Comment:
This is maybe okay, but not _super_ beautiful. A nicer solution would be to create an enum with cases `Dynamic`, `Crescendo`, `Decrescendo`, and add a field of that enum type to the `struct`/`QVariantMap` mentioned in my previous comment. 

Exposing enums from C++ to QML is done using the `Q_ENUM` macro; you can search the codebase for examples. 

(The idea of this suggestion and my previous one is to keep related information closer together; this will make the code easier to understand and safer to maintain because you don't risk updating one place and forgetting another.)